TestProject.BL/Services/PostService.cs

```diff
@@ -30,7 +30,7 @@ public class PostService : IPostService
         public async Task Create(EditPostModel editPostModel, string userEmail)
         {
             var post = _editPostMapper.ToDalModel(editPostModel);
-            var user = await _userRepository.GetUserByEmail(userEmail);
+            var user = _userRepository.Get(u => u.Email == userEmail).First();
```

Comment:
It is used 7 times in the commit with the same predicate. Worth extracting to an extension method.